services/diddoc/diddoc_query.go

```diff
@@ -17,31 +15,27 @@ import (
 
 type QueryDIDDocRequestService struct {
 	services.BaseRequestService
+	Profile      string
 	FirstHandler queries.BaseQueryHandlerI
 }
 
 func (dd *QueryDIDDocRequestService) Setup(c services.ResolverContext) error {
 	return nil
 }
 
+// function to prepare the Query DIDDoc Request with specific conditions
 func (dd *QueryDIDDocRequestService) SpecificPrepare(c services.ResolverContext) error {
 	if dd.AreDidResolutionQueries(c) {
 		dd.IsDereferencing = false
 	} else {
-		// if resource queries are provided and the profile is W3IDDIDRES then dereferencing is false
```

Comment:
Removing this check here breaks W3DIDRES responses.

Reply:
@DaevMithran  This seems duplicate, I have reinstated the check in `BaseQueryHandler SetNext()`, as the BaseQueryHandler is initialized everytime for a new handler.